### PR TITLE
Improve and fix bugs in runner management page

### PR DESCRIPTION
**Pull request by yp05327**
Changes:
- Add no permission to edit tooltip
image
- fix the logic in ActionRunner.Editable
  -  Gitea admin user can edit runner in org/repo runner settings page
      image
  - Org owner can edit org/repo runner in org's repo runner settings page
     image
  - org admin can only edit org's repo runner
     image
- add missed locale word: `runners.task_list.no_tasks`
- ~~fix overflow of `Recent tasks on this runner`~~ This has been fixed.
Before:
image
After:
image

### CODE REVIEW

**Comment by silverwind** [comment]
image

Edit icon looks slighly misaligned with cross icon horizontally.

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -3382,6 +3382,7 @@ runners.owner_type = Type
 runners.description = Description
 runners.labels = Labels
 runners.last_online = Last Online Time
+runners.no_permission_to_edit = You have no permission to edit this runner.
```
Comment (silverwind) [inline options/locale/locale_en-US.ini]:
```suggestion
runners.no_permission_to_edit = You don't have permission to edit this runner.
```

**Comment by yp05327** [comment]
> image
> 
> Edit icon looks slighly misaligned with cross icon horizontally.

Done.

File: web_src/css/runner.css [hunk 2]
```diff
@@ -10,6 +10,10 @@
   margin-left: 0.5em;
 }
 
+.runner-container .runner-ops > div {
```
Comment (silverwind) [inline web_src/css/runner.css]: `div` seems too generic of a selector. It's not obvious what the rule does when looking at it. Maybe just add a `gt-ml-3` via HTML instead?

Comment (yp05327) [reply web_src/css/runner.css]: Done.

File: models/actions/runner.go [hunk 3]
```diff
@@ -94,14 +119,40 @@ func (r *ActionRunner) AllLabels() []string {
...
+		return true, nil
+	}
+
+	if repo != nil {
+		// repo runner in repo runners list
+		if r.Type() == RunnerTypeRepository && r.RepoID == repo.ID {
+			return true, nil
+		}
+
+		if err := repo.LoadOwner(db.DefaultContext); err != nil {
```
Comment (wolfogre) [inline models/actions/runner.go]: We should avoid using `db.DefaultContext`

**Review by wolfogre — COMMENTED** [review]
After this, it will be possible to edit a global/org runner on the settings page of a repo if the doer has permission. It doesn't look really good to me.

image
image

Just my opinion, I would say it could be better to redirect to org/admin settings page.

File: models/actions/runner.go (hunk 3 above)
Comment (Gabriel-Gotti) [reply models/actions/runner.go:138]: db.DefaultContext is the err. ive been in and out of my system and thats the real overight

Comment (yp05327) [reply models/actions/runner.go]: Done.

**Comment by denyskon** [comment]
@yp05327 Is this PR still relevant? If so, could you update it?

**Comment by yp05327** [comment]
I'll try to update it.

**Comment by yp05327** [comment]
Added individual runner support.

**Comment by yp05327** [comment]
Update is done.

**Comment by lunny** [comment]
> After this, it will be possible to edit a global/org runner on the settings page of a repo if the doer has permission. It doesn't look really good to me.
> 
> image
> 
> image
> 
> Just my opinion, I would say it could be better to redirect to org/admin settings page.

It seems this hasn't been followed?

**Comment by denyskon** [comment]
Isn't the second option used currently? Looks correct to me, the admin should always use admin routes if possible

**Pull request closed without merging**